### DISCUSSION TITLE
How to bundle external types into own package?

### DISCUSSION THREAD

**Discussion by moltar** [egoist/tsup#989] [General]
Let's say I want to re-package and re-publish some types from an external npm package, but not have it be in the `dependencies`.

The solution on the TS code side is quite easy:

```ts
// ./index.ts

export type { Config as RenovateConfig } from 'renovate'
```

But how do I configure `tsup` to bundle the types, rather than import them in the final built asset?

In effect, this is like bundling all third-party code. I know `esbuild` is capable of that. Just not sure how to do it in `tsup`.

More details:

The use case is that I want to create a helper package for a third-party package, but do not want to have to force the user to install the _actual_ third-party package, because it's [very heavy](https://i.redd.it/tfugj4n3l6ez.png) (10 MB). I am talking about [`renovate`](https://www.npmjs.com/package/renovate), which is in effect a whole application. I am working on a tool that helps manage Renovate configuration, which can be shared, but it does not need the entire `renovate` baggage. Most users will use the Renovate Bot GitHub Action anyway and just need a single `.json` file in the end.

**Comment by OmGuptaIND** [comment]
any updates on this @moltar. Trying to do the same thing

**Reply by moltar** [reply]
@OmGuptaIND This config will do it.

```json
  "dts": {
    "resolve": true
  },
```

You also want the package to be a devDep installed, otherwise `tsup` will ignore it, and not bundle it.

But ... I did run into a bug somewhere in the chain, where this may not work for all cases. I have a repro case: https://github.com/moltar/tsup-dts-resolve
